### ISSUE TITLE
ReturnDefaultIfNotRegistered is not respected between scopes

### ISSUE THREAD

**Issue by rick-palmsens** [dadhi/DryIoc#434]
The following code will throw even though `IfUnresolved` is set to `ReturnDefaultIfNotRegistered`:
```csharp
class Program
{
    static void Main(string[] args)
    {
        var container = new Container();

        using (var scope = container.OpenScope())
        {
            var foo = scope.Resolve<Foo>(IfUnresolved.ReturnDefaultIfNotRegistered); // Does not throw
            scope.UseInstance<Foo>(new Foo()); // Do a registration within the scope
        }

        using (var scope = container.OpenScope())
        {
            var foo = scope.Resolve<Foo>(IfUnresolved.ReturnDefaultIfNotRegistered); // Throws
        }
    }
}
```
Since the registration happened within the first scope, I'd expect the second scope to simple be empty and not throw. Looking at the stack trace, the `IfUnresolved` parameter seems to get discarded half way through, and the code simply throws instead of checking if it should actually throw.

**Comment by dadhi** [comment]
@rick-palmsens Thanks for finding, will check. The fix PRs are appreciated as well (as you have already looking into the stack trace) ;)

**Comment by dadhi** [comment]
I have fixed the issue in the upcoming v4.8.4 but going forward please use `Use` instead of `UseInstance` as the later will be removed in the next major version.
